source/rules/no-nested-subscribe.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/eslint-plugin-rxjs
+ */
+
+import { Rule } from "eslint";
+import * as es from "estree";
+import { couldBeType } from "tsutils-etc";
+import { getTypeCheckerAndNodeMap } from "../utils";
+const esquery = require("esquery");
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      category: "RxJS",
+      description:
+        "Disallows the calling of `subscribe` within a `subscribe` callback.",
+      recommended: true
+    },
+    fixable: null,
+    messages: {
+      forbidden: "Nested subscribe calls are forbidden."
+    },
+    schema: []
+  },
+  create: context => {
+    const { nodeMap, typeChecker } = getTypeCheckerAndNodeMap(context);
+    const subscribeQuery =
+      "CallExpression > MemberExpression[property.name='subscribe']";
+
+    return {
+      [subscribeQuery]: (node: es.MemberExpression) => {
+        const identifier = nodeMap.get(node.object);
+        const identifierType = typeChecker.getTypeAtLocation(identifier);
+
+        if (!couldBeType(identifierType, "Observable")) {
+          return;
+        }
+
+        const callExpression = (node as any).parent as es.CallExpression;
```

Comment:
If the ES AST nodes have an internal - i.e. not typed - `parent`, I think we should create a utility function to get it, so that we don't have to have `any` assertions sprinkled about the place:

```ts
function getParent(node: es.Node): es.Node;
```

I'd be happy for the return type to assume there is always a parent - i.e. it's never going to be `undefined` - as that's the way it in in TS.

Reply:
👍 I'll make those changes.
From what I can see, the experimental lib also has a parent node.